src/parser/statement.js

```diff
@@ -530,12 +530,11 @@ pp.parseFor = function (node, init) {
 // same from parser's perspective.
 
 pp.parseForIn = function (node, init, forAwait) {
-  let type;
+  const type = this.match(tt._in) ? "ForInStatement" : "ForOfStatement";
   if (forAwait) {
+    node.await = true;
     this.eatContextual("of");
-    type = "ForAwaitStatement";
   } else {
```

Comment:
Should we set the `node.await` to `false` there? According to the changes in the spec I would expect a `boolean`.

Since `undefined` is falsy this isn't an issue.